kauri/aggduration.go

```diff
@@ -0,0 +1,68 @@
+package kauri
+
+import (
+	"slices"
+	"time"
+
+	"github.com/relab/hotstuff"
+	"github.com/relab/hotstuff/internal/latency"
+	"github.com/relab/hotstuff/internal/tree"
+	"github.com/relab/hotstuff/modules"
+)
+
+// AggregationLatency calculates the maximum duration of a path in the tree.
+type AggregationLatency struct {
```

Comment:
@hanish520 Wouldn't it be more suitable to keep this in the `tree` package? The aggregation latency is valid for many tree configurations, not just Kauri, no?